### ISSUE TITLE
[Bug report] Web fails to load since 2.0.2

### ISSUE THREAD

**Issue by vixez** [fluttercandies/flutter_image_compress#244]
### Version

2.0.2

### Platforms

Web

### Device Model

Chrome

### flutter info

```sh
[!] Flutter (Channel stable, 3.10.0, on macOS 13.2.1 22D68 darwin-x64, locale en-GB)
    • Flutter version 3.10.0 on channel stable at /Users/*/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (2 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/glennruysschaert/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • ANDROID_HOME = /Users/glennruysschaert/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.75.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (3 available)
    • Glenn’s iPad (mobile) • 00008101-000D4D112139003A • ios            • iOS 16.4.1 20E252
    • macOS (desktop)       • macos                     • darwin-x64     • macOS 13.2.1 22D68 darwin-x64
    • Chrome (web)          • chrome                    • web-javascript • Google Chrome 113.0.5672.92

[✓] Network resources
    • All expected network resources are available.
```

### How to reproduce?

Running the app on iOS and Android works. Web throws `Unexpected null value`, presumably at `packages/flutter_image_compress_web/flutter_image_compress_web.dart 14:27`

I currently reverted back to `1.1.3`.

### Logs

```sh
Error: Unexpected null value.
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 288:49      throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 606:63  nullCheck
packages/flutter/src/rendering/binding.dart 303:61                                createViewConfiguration
packages/flutter/src/rendering/binding.dart 215:44                                initRenderView
packages/flutter/src/rendering/binding.dart 41:5                                  initInstances
packages/flutter/src/widgets/binding.dart 263:11                                  initInstances
packages/flutter/src/foundation/binding.dart 151:5                                new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1309:3                                  new
packages/flutter/src/widgets/binding.dart 1306:7                                  ensureInitialized
packages/flutter_image_compress_web/flutter_image_compress_web.dart 14:27         registerWith
web_plugin_registrant.dart 52:27                                                  registerPlugins
web_entrypoint.dart 27:24                                                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
lib/ui/initialization.dart 73:24                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5                _async
lib/ui/initialization.dart 71:16                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 431:10  callMethod
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 435:5   dsend
lib/_engine/engine/app_bootstrap.dart 31:18                                       autoStart
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50                <fn>
dart-sdk/lib/async/zone.dart 1661:54                                              runUnary
dart-sdk/lib/async/future_impl.dart 147:18                                        handleValue
dart-sdk/lib/async/future_impl.dart 784:44                                        handleValueCallback
dart-sdk/lib/async/future_impl.dart 813:13                                        _propagateToListeners
dart-sdk/lib/async/future_impl.dart 584:5                                         [_completeWithValue]
dart-sdk/lib/async/future.dart 531:22                                             <fn>
dart-sdk/lib/async/zone.dart 1661:54                                              runUnary
dart-sdk/lib/async/future_impl.dart 147:18                                        handleValue
dart-sdk/lib/async/future_impl.dart 784:44                                        handleValueCallback
dart-sdk/lib/async/future_impl.dart 813:13                                        _propagateToListeners
dart-sdk/lib/async/future_impl.dart 584:5                                         [_completeWithValue]
dart-sdk/lib/async/future.dart 531:22                                             <fn>
dart-sdk/lib/async/zone.dart 1661:54                                              runUnary
dart-sdk/lib/async/future_impl.dart 147:18                                        handleValue
dart-sdk/lib/async/future_impl.dart 784:44                                        handleValueCallback
dart-sdk/lib/async/future_impl.dart 813:13                                        _propagateToListeners
dart-sdk/lib/async/future_impl.dart 584:5                                         [_completeWithValue]
dart-sdk/lib/async/future_impl.dart 657:7                                         callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                                  _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                                   _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 177:15               <fn>
```

### Example code (optional)

```
run() async {
  await runZonedGuarded(() async {
    runApp(MyApp());
}
```

**Comment by vixez** [comment]
Looks like this is related to `runZonedGuarded`

```
run() async {
  await runZonedGuarded(() async {
    runApp(MyApp());
}
```

**Comment by mcagrigoktas** [comment]
Same issue 

packages/flutter/src/widgets/binding.dart 1306:7                                  ensureInitialized
packages/flutter_image_compress_web/flutter_image_compress_web.dart 14:27         registerWith
web_plugin_registrant.dart 52:27                                                  registerPlugins
web_entrypoint.dart 27:24                                                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
lib/ui/initialization.dart 73:24                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5                _async
lib/ui/initialization.dart 71:16                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 431:10  callMethod
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 435:5   dsend
lib/_engine/engine/app_bootstrap.dart 62:22                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5                _async
lib/_engine/engine/app_bootstrap.dart 60:25                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 249:16                    callConstructor
lib/_engine/engine/app_bootstrap.dart 58:14                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
. 106:28                                                                          <fn>

**Comment by rvndsngwn** [comment]
```
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...
This app is linked to the debug service: ws://127.0.0.1:59096/CxeSdcUetIM=/ws
Debug service listening on ws://127.0.0.1:59096/CxeSdcUetIM=/ws
Debug service listening on ws://127.0.0.1:59096/CxeSdcUetIM=/ws
Error: Unexpected null value.
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 288:49      throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 606:63  nullCheck
packages/flutter/src/rendering/binding.dart 303:61                                createViewConfiguration
packages/flutter/src/rendering/binding.dart 215:44                                initRenderView
packages/flutter/src/rendering/binding.dart 41:5                                  initInstances
packages/flutter/src/widgets/binding.dart 263:11                                  initInstances
packages/flutter/src/foundation/binding.dart 151:5                                new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1189:37                                 new
packages/flutter/src/widgets/binding.dart 1309:3                                  new
packages/flutter/src/widgets/binding.dart 1306:7                                  ensureInitialized
packages/flutter_image_compress_web/flutter_image_compress_web.dart 14:27         registerWith
web_plugin_registrant.dart 36:27                                                  registerPlugins
web_entrypoint.dart 27:24                                                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
lib/ui/initialization.dart 73:24                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5                _async
lib/ui/initialization.dart 71:16                                                  <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 431:10  callMethod
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 435:5   dsend
lib/_engine/engine/app_bootstrap.dart 62:22                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5                _async
lib/_engine/engine/app_bootstrap.dart 60:25                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 249:16                    callConstructor
lib/_engine/engine/app_bootstrap.dart 58:14                                       <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 367:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 372:39  dcall
. 53:26                                                                           <fn>

```

**Comment by CaiJingLong** [comment]
The error just happen in Flutter 3.10.0 ?

**Comment by CaiJingLong** [comment]
I committed https://github.com/fluttercandies/flutter_image_compress/commit/c943c19d0c7816b1dadeea7532245421cd393746.
Wait for release new version.

**Comment by CaiJingLong** [comment]
Released new version 2.0.3

**Comment by vixez** [comment]
Thanks for the quick response!
